node/src/isolate/external.cpp

```diff
@@ -362,6 +397,22 @@ spawn_load_t::apply() {
 }
 
 void
+metrics_load_t::apply() {
+    auto _inner = inner;
+    stream.apply([&](decltype(stream.unsafe())& stream){
```

Comment:
I don't think that aquiring a mutex for code consistency is a good idea) other handles can be cancelled from other thread, and it looks like this one can not.